test/test_support.ts

```diff
@@ -143,3 +147,178 @@ export function goldenTests(): GoldenFileTest[] {
 
   return tests;
 }
+
+
+export function getLineAndColumn(source: string, token: string): {line: number, column: number} {
+  const lines = source.split('\n');
+  const line = lines.findIndex(l => l.indexOf(token) !== -1) + 1;
+  if (line === 0) {
+    throw new Error(`Couldn't find token '${token}' in source`);
+  }
+  const column = lines[line - 1].indexOf(token);
+  return {line, column};
+}
+
+export interface CompilerOptions {
+  useTransformer: boolean;
+  outFile?: string;
+  filesNotToProcess: Set<string>;
+  inlineSourceMap: boolean;
+  generateDTS: boolean;
+}
+
+const DEFAULT_COMPILER_OPTIONS = {
+  filesNotToProcess: new Set<string>(),
+  inlineSourceMap: false,
+  generateDTS: false,
+};
+
+export function compile(
+    sources: Map<string, string>,
+    partialOptions: Partial<CompilerOptions>&{useTransformer: boolean}): {
+  compiledJS: string,
+  dts: string | undefined,
+  sourceMap: SourceMapConsumer,
+  sourceMapText: string
+} {
+  const options: CompilerOptions = {...DEFAULT_COMPILER_OPTIONS, ...partialOptions};
+  const resolvedSources = new Map<string, string>();
+  for (const fileName of toArray(sources.keys())) {
+    resolvedSources.set(ts.sys.resolvePath(fileName), sources.get(fileName)!);
+  }
+
+  let compilerOptions: ts.CompilerOptions;
+  if (options.inlineSourceMap) {
+    compilerOptions = {
+      inlineSourceMap: options.inlineSourceMap,
+      inlineSources: true,
+      outFile: options.outFile,
+      experimentalDecorators: true,
+      declaration: options.generateDTS,
+    };
+  } else {
+    compilerOptions = {
+      sourceMap: true,
+      inlineSources: true,
+      outFile: options.outFile,
+      experimentalDecorators: true,
+      declaration: options.generateDTS,
+    };
+  }
+
+  const fileNames = toArray(sources.keys());
+
+  const tsickleHost: tsickle.TsickleHost&tsickle.TransformerHost = {
+    shouldSkipTsickleProcessing: (fileName) =>
+        fileNames.indexOf(fileName) === -1 || options.filesNotToProcess.has(fileName),
+    pathToModuleName: cliSupport.pathToModuleName,
+    shouldIgnoreWarningsForPath: (filePath) => false,
+    fileNameToModuleId: (fileName) => fileName,
+  };
+
+  const {jsFiles, diagnostics} = options.useTransformer ?
+      compileWithTransfromer(resolvedSources, fileNames, tsickleHost, compilerOptions) :
+      compileWithTsickleCompilerHost(resolvedSources, fileNames, tsickleHost, compilerOptions);
+
+  if (diagnostics.length) {
+    console.error(tsickle.formatDiagnostics(diagnostics));
+    assert.fail();
+    return {compiledJS: '', dts: '', sourceMap: sourceMapTextToConsumer(''), sourceMapText: ''};
+  }
+
+  let outFile = options.outFile;
```

Comment:
(New comment thread because you overwrote the commits of the old thread, so I cannot continue there)

If your code uses ES6 modules, `tsc` complains because it cannot (easily) combine modules into a single output file, so that's reasonable on that side.

Regarding `if (!outFile) {`, things to fix:
* [ ] please change the code to use a different variable name that better indicates what's going on. You're currently overloading the meaning of `outFile` to be both the file produced by tsc and the file you want to extract the output for, so this variable here should have a different name. 
* [ ] the implicit behaviour to combine inputs is non-obvious, this should have implementation documentation here. You're using `outFile` as a convenient way to get only one `compiledJS` output. That's a bit misleading, as in normal operation tsickle will generate individual files. Tests would arguably be more understandable if you returned all generated outputs and asserted on that. I understand the difference in convenience (though it might be worth considering if the alternative is so much more verbose), but this at least needs docs.
* [ ] you seem to have a bunch of tests that set outFile despite only passing a single input - that doesn't make sense then, does it? IIUC `outFile` should only be set when you pass multiple inputs and want to test composition of source maps in that situation.